Update Heroku buildpack support from 0.5.0.0 to v7.0.1

yingw787 opened the pull request:
Hello,

I had wanted to use the latest version of PostgREST, but deploying from Heroku had a max supported version number of 0.5.0.0. I saw in the documentation Heroku was the preferred deployment platform, so I upgraded the Heroku buildpack to support v7.0.1.

Primary issue is changing the ubuntu.tar.gz to linux-x64-static.tar.gz, in addition I also removed the cache libs because it supported 0.4.x only, and updated the documentation.

Thanks
Ying

yingw787 commented:
@steve-chavez can you please take a look at this pull request, add any contributors / reviewers if necessary, and if possible merge in this update? I think it may be helpful to others if they can deploy PostgREST::latest.

steve-chavez commented on bin/compile:
```diff
@@ -48,36 +38,12 @@ if [[ $CLEAR_CACHE -eq 1 ]] ; then
 
-############# Library files ###############################
-
-if [ ! -e $CACHE_DIR/ghc-libs ]; then
```
@yingw787 Hi there. Instead of deleting these lines, could you add a conditional(< 6.0.0) for downloading library files so previous versions keep working? I think some users have deployments with the old versions and it would be good to keep backwards compat.

The change related to the binary looks good.

steve-chavez commented on bin/compile:
```diff
@@ -48,36 +38,12 @@ if [[ $CLEAR_CACHE -eq 1 ]] ; then
 
-############# Library files ###############################
-
-if [ ! -e $CACHE_DIR/ghc-libs ]; then
-  LIBS_URL="https://github.com/PostgREST/postgrest-heroku/releases/download/v0.4.4.0/ghc-libs.tar.xz"
-  if curl --output /dev/null --silent --head --fail $LIBS_URL; then
-    echo "-----> Downloading and caching libraries for PostgREST"
```
@yingw787 Hi there. Instead of deleting these lines, could you add a conditional(< 6.0.0) for downloading library files so previous versions keep working? I think some users have deployments with the old versions and it would be good to keep backwards compatibility.

steve-chavez commented on bin/compile:
```diff
@@ -48,36 +38,12 @@ if [[ $CLEAR_CACHE -eq 1 ]] ; then
-
 if [ ! -e $CACHE_DIR/$BIN ]; then
-  BIN_URL="https://github.com/PostgREST/postgrest/releases/download/v${POSTGREST_VER}/postgrest-${RELEASE_VER}-ubuntu.tar.xz"
+  BIN_URL="https://github.com/PostgREST/postgrest/releases/download/v${POSTGREST_VER}/postgrest-${RELEASE_VER}-linux-x64-static.tar.xz"
```
A conditional for >= 6.0.0 to change to the linux-x64-static bin would also do here, I think.

steve-chavez commented on README.md:
```diff
@@ -2,38 +2,75 @@
-the **Virginia region** because that's where Heroku's dynos are
-and it will decrease latency between the server and db.
+We recommend spinning up a Postgres database with
+[ElephantSQL](https://www.elephantsql.com/) for free tier usage.
```
How about making ElephantSQL an alternative to RDS? You could put it below on another subheading.

I think RDS should stay because most users are familiar with it for production.

yingw787 commented:
@steve-chavez hey man, thanks for the comments; haven't gotten around to addressing your comments but rest assured I should be able to do it by end of this weekend. Are there specific versions of PostgREST / configurations I should test? Would be great to have a CI/CD test runner, though if that doesn't exist I can speclatively execute assumed manual tests.

steve-chavez commented:
@yingw787 Sorry for the late reply. I think you would only need to test the old `5.2.0` version and then any of the newer versions(latest would be fine). That would ensure the conditional for `< 6.0.0 <=` works.

Manual tests would be good(I'll also test). This repo is rarely changed, so I haven't had the need of setting a CI.

steve-chavez commented:
Superseded by https://github.com/PostgREST/postgrest-heroku/pull/33. Many thanks for your help here @yingw787
